test/unit/org/apache/cassandra/auth/ResourcesTest.java

```diff
@@ -0,0 +1,78 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.auth;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+import org.junit.Test;
+
+import org.apache.cassandra.db.marshal.DoubleType;
+import org.apache.cassandra.db.marshal.Int32Type;
+
+import static org.junit.Assert.assertEquals;
+
+public class ResourcesTest
+{
+
+    @Test
+    public void testResourceFromName()
+    {
+        // role resources
+        assertEquals(RoleResource.root(), Resources.fromName("roles"));
+        assertEquals(RoleResource.role("role1"), Resources.fromName("roles/role1"));
+
+        // data resources
+        assertEquals(DataResource.root(), Resources.fromName("data"));
+        assertEquals(DataResource.keyspace("ks1"), Resources.fromName("data/ks1"));
+        assertEquals(DataResource.table("ks1", "t1"), Resources.fromName("data/ks1/t1"));
+
+        // function resources
+        assertEquals(FunctionResource.root(), Resources.fromName("functions"));
+        assertEquals(FunctionResource.keyspace("ks1"), Resources.fromName("functions/ks1"));
+        assertEquals(FunctionResource.function("ks1", "f1", Collections.emptyList()),
+                Resources.fromName("functions/ks1/f1[]"));
+        // this is actually supported by an explicit check in TypeParser
+        assertEquals(FunctionResource.function("ks1", "f1", Arrays.asList(Int32Type.instance, DoubleType.instance)),
+                Resources.fromName("functions/ks1/f1[Int32Type^DoubleType]"));
+        assertEquals(FunctionResource.function("ks1", "f1", Arrays.asList(Int32Type.instance, DoubleType.instance)),
+                Resources.fromName("functions/ks1/f1[org.apache.cassandra.db.marshal.Int32Type^org.apache.cassandra.db.marshal.DoubleType]"));
+    }
+
+    @Test
+    public void testResourceGetName()
+    {
+        // role resources
+        assertEquals("roles", RoleResource.root().getName());
+        assertEquals("roles/role1", RoleResource.role("role1").getName());
+
+        // data resources
+        assertEquals("data", DataResource.root().getName());
+        assertEquals("data/ks1", DataResource.keyspace("ks1").getName());
+        assertEquals("data/ks1/t1", DataResource.table("ks1", "t1").getName());
+
+        // function resources
+        assertEquals("functions", FunctionResource.root().getName());
+        assertEquals("functions/ks1", FunctionResource.keyspace("ks1").getName());
+        assertEquals("functions/ks1/f1[]",
+                FunctionResource.function("ks1", "f1", Collections.emptyList()).getName());
+        assertEquals("functions/ks1/f1[org.apache.cassandra.db.marshal.Int32Type^org.apache.cassandra.db.marshal.DoubleType]",
+                FunctionResource.function("ks1", "f1", Arrays.asList(Int32Type.instance, DoubleType.instance)).getName());
+    }
+}
```

Comment:
Missing new line

Reply:
Added. I just did not know it is a part of the project preferred code style.